psydac/api/basic.py

```diff
@@ -22,20 +22,54 @@
 #==============================================================================
 # TODO have it as abstract class
 class BasicCodeGen:
-    """ Basic class for any discrete concept that needs code generation """
+    """ Basic class for any discrete concept that needs code generation.
 
-    def __init__(self, expr, **kwargs):
+    Parameters
+    ----------
+
+    folder: str
+        The output folder where we generate the code.
+
+    comm: MPI.Comm
+        The mpi communicator used in the paralle case.
+
+    root: int
+        The process that is responsible of generating the code.
+
+    discrete_space: FemSpace | list of FemSpace
+        The discrete fem spaces.
+
+    kernel_expr : sympde.expr.evaluation.KernelExpression
+        The atomic representation of the bi-linear form.
+
+    quad_order: list of tuple
+        The number of quadrature points used in the assembly method.
+
+    is_rational_mapping : bool
+        takes the value of True if the mapping is rational.
+
+    mapping: Sympde.topology.Mapping
+        The symbolic mapping of the bi-linear form domain.
+
+    mapping_space: FemSpace
+       The discete space of the mapping.
+
+    num_threads: int
+       Number of threads used in the computing kernels.
+ 
+    backend: dict
+        The backend used to accelerate the computing kernels.
```

Comment:
Can you please add a sentence saying what this dictionary should contain? It is probably enough to point to the relevant file: [psydac/api/settings.py](https://github.com/pyccel/psydac/blob/devel/psydac/api/settings.py).

Reply:
fixed in [797bf9e](https://github.com/pyccel/psydac/pull/232/commits/797bf9eb03c4f2f616a22fa7b86083b68c8809e9)